echo_server.py

```diff
@@ -29,5 +33,39 @@ def start_server():
     finally:
         raise
 
+
+def response_ok(uri):
+    okay_response = "HTTP/1.1 200 OK\r\n"
+    headers = "Content-Type: text/plain\r\n"
+    body = "you requested: {}".format(uri)
+    return "{}{}{}\r\n".format(okay_response, headers, body)
+
+
+def response_error(error_num, error_msg):
+    error_response = "HTTP/1.1 {} ERROR\r\n".format(error_num)
+    headers = "Content-Type: text/plain\r\n"
+    body = "ERROR {}, {}\r\n".format(error_num, error_msg)
+    return "{}{}{}".format(error_response, headers, body)
```

Comment:
Again, you're missing the blank line that must separate the headers in the HTTP response from the body of the response.